services/web/server/src/simcore_service_webserver/catalog_utils.py

```diff
@@ -0,0 +1,177 @@
+import logging
+from dataclasses import dataclass
+from typing import Optional
+
+from aiohttp import web
+from models_library.services import BaseServiceIOModel, ServiceInput, ServiceOutput
+from pint import PintError, UnitRegistry
+from yarl import URL
+
+from . import catalog_client
+from ._constants import RQ_PRODUCT_KEY, X_PRODUCT_NAME_HEADER
+from .catalog_client import to_backend_service
+from .catalog_settings import get_plugin_settings
+from .login.decorators import RQT_USERID_KEY, login_required
+from .security_decorators import permission_required
+
+logger = logging.getLogger(__name__)
+
+
+@login_required
+@permission_required("services.catalog.*")
+async def reverse_proxy_handler(request: web.Request) -> web.Response:
+    """
+        - Adds auth layer
+        - Adds access layer
+        - Forwards request to catalog service
+
+    SEE https://gist.github.com/barrachri/32f865c4705f27e75d3b8530180589fb
+    """
+    user_id = request[RQT_USERID_KEY]
+    settings = get_plugin_settings(request.app)
+
+    # path & queries
+    backend_url = to_backend_service(
+        request.rel_url,
+        URL(settings.base_url),
+        settings.CATALOG_VTAG,
+    )
+    # FIXME: hack
+    if "/services" in backend_url.path:
+        backend_url = backend_url.update_query({"user_id": user_id})
```

Comment:
shall we ask for user_id everywhere? or we could pass in the parameters?

Reply:
By parameters you mean in the path? Let's talk about it offline ... i tis a bit long to write.